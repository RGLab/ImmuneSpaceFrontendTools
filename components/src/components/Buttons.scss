
.df-highlighted-button {
    margin: 9px 5px;
    background-color: #de7e3a;
    border-radius: 2px;
    color: white;
    padding-top: 0;
    height: 20px;
    
    &:hover {
      border: 1px solid #cf6319;
      color: white;
    }
  }


.df-row-of-buttons {
    display: inline-block;
    width: 100%
  }
  