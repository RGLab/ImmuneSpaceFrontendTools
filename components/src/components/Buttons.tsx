import * as React from 'react';
import "./Buttons.scss"

interface HighlightedButtonProps {
    action?: () => void;
    href?: string;
    id?: string;
    disabled?: boolean
}

interface RowOfButtonProps {
    id?: string;
}


// Button or link with highlighted style
export const HighlightedButton: React.FC<HighlightedButtonProps> = ({href, action, id, disabled, children}) => {
    return <a href={href} className={"btn df-highlighted-button" + (disabled ? " disabled" : "")} onClick={action} id={id}>
        {children}
    </a>;
  };

// export const HighlightedButton = React.memo(HighlightedButtonFC)


// Floats all children in a nice little row
export const RowOfButtons: React.FC<RowOfButtonProps> = ({children, id}) => {
    return <div className={"df-row-of-buttons"} id={id}>
    {React.Children.map(children || null, (child, i) => {
        return (
            <div style={{float: "left", padding: "3px 10px"}}>
                {child}
            </div>
            )
        })}
    </div>
}