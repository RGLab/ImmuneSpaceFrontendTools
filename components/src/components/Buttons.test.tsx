import React from 'react'
import * as renderer from 'react-test-renderer'
import * as buttons from './Buttons'
import { mount } from "enzyme";

describe("<HighlightedButton />", () => {
    test("render with all props", () => {
        const tree = renderer.create(<buttons.HighlightedButton label="Click me!" action={jest.fn()} id={"this-id"}/>).toJSON()
        expect(tree).toMatchSnapshot()  
    })
    test("change text", () => {
        const button = mount(<buttons.HighlightedButton action={jest.fn()} label={"Before"} />)
        expect(
            button.text()
          ).toEqual('Before')

        button.setProps({label: "After"})
        expect(
            button.text()
        ).toEqual("After")
    })
    test("id", () => {
        const button = mount(<buttons.HighlightedButton action={jest.fn()} label={"Button"} id={"test-button"} />)
        expect(button.find("#test-button").hostNodes().text()).toEqual("Button")
    })
    test("click", () => {
        const onClick = jest.fn()
        const button = mount(<buttons.HighlightedButton action={onClick} label={"Apply"}/>);
        expect(onClick.mock.calls.length).toBe(0)
        button.simulate("click")
        expect(onClick.mock.calls.length).toBe(1)
    })
    test("href", () => {
        const button =  mount(<buttons.HighlightedButton href={"go-to-site"} label={"Apply"}/>);
        expect(button.find('a[href="go-to-site"]').text()).toEqual("Apply")
    })
})


