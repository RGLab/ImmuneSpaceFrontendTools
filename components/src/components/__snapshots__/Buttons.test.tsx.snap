// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<HighlightedButton /> render with all props 1`] = `
<a
  className="btn df-highlighted-button"
  id="this-id"
  onClick={[MockFunction]}
/>
`;
